apps/chat/bots.py

```diff
@@ -103,19 +104,29 @@ def _call_predict(self, input_str, save_input_to_history=True, attachments: list
             tag, chain = self._get_child_chain(input_str, attachments)
         else:
             tag, chain = None, self.chain
+
         result = chain.invoke(
             input_str,
             config={
                 "configurable": {
                     "save_input_to_history": save_input_to_history,
+                    "save_output_to_history": self.terminal_chain is None,
                     "experiment_tag": tag,
                 }
             },
             attachments=attachments,
         )
 
         if self.terminal_chain:
-            result = self.terminal_chain.invoke(result.output)
+            result = self.terminal_chain.invoke(
```

Comment:
It feels wrong to not save the output/input, but monitoring tools will pick the inputs/outputs up when we have it